LibreHardwareMonitorLib/Hardware/Controller/Nzxt/GridV3.cs

```diff
@@ -0,0 +1,180 @@
+﻿// This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
+// If a copy of the MPL was not distributed with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+// Copyright (C) LibreHardwareMonitor and Contributors.
+// All Rights Reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using HidSharp;
+
+namespace LibreHardwareMonitor.Hardware.Controller.Nzxt
+{
+    /**
+     * Support for the GRID+ V3 devices from NZXT
+     */
+    internal sealed class GridV3 : Hardware
+    {
+        // Some initialization messages to send to the controller. No visible effects but NZXT CAM send them.
+        private static readonly byte[] _initialize1 = { 0x01, 0x5c };
+        private static readonly byte[] _initialize2 = { 0x01, 0x5d };
+        private static readonly byte[] _initialize3 = { 0x01, 0x59 };
+        private readonly byte[] _setFanSpeedMsg;
+
+        private readonly HidStream _stream;
+        private readonly Dictionary<int, byte[]> _rawData = new Dictionary<int, byte[]>();
+
+        private const int FansCount = 6;
+        private readonly Sensor[] _rpmSensors = new Sensor[FansCount];
+        private readonly Sensor[] _voltages = new Sensor[FansCount];
+        private readonly Sensor[] _currents = new Sensor[FansCount];
+        private readonly Sensor[] _powers = new Sensor[FansCount];
+        private readonly Sensor[] _pwmControls = new Sensor[FansCount];
+        private readonly Sensor _noise;
+
+        private readonly Control[] _fanControls = new Control[FansCount];
```

Comment:
Format this block

Reply:
Don't understand what should I do. I checked other sources in LHM, looks acceptable.

Reply:
Format document in VS (it should sort these fields).

Reply:
format did nothing. Sorted manually